graphql/core/validation/rules.py

```diff
@@ -591,7 +591,54 @@ def bad_value_for_default_arg_message(var_name, type, value):
 
 
 class VariablesInAllowedPosition(ValidationRule):
-    pass
+    visit_spread_fragments = True
+
+    def __init__(self, context):
+        super(VariablesInAllowedPosition, self).__init__(context)
+        self.var_def_map = {}
+        self.visited_fragment_names = set()
+
+    def enter_OperationDefinition(self, *args):
+        self.var_def_map = {}
+        self.visited_fragment_names = set()
+
+    def enter_VariableDefinition(self, var_def_ast, *args):
+        self.var_def_map[var_def_ast.variable.name.value] = var_def_ast
+
+    def enter_Variable(self, variable_ast, *args):
+        var_name = variable_ast.name.value
+        var_def = self.var_def_map[var_name]
```

Comment:
Just read this comment, but essentially same as what I said in https://github.com/dittos/graphqllib/pull/38#issuecomment-141724009 !